client/src/main/java/com/sap/mobile/services/client/push/PushPayload.java

```diff
@@ -102,14 +105,28 @@ public Builder priority(String priority) {
 		/**
 		 * The data property is a serialized / escape JSON key/value map. The object is
 		 * not shown in the notification, but can be used by the mobile app notification
-		 * handler. Android only support maps/key value pairs and other JSON objects are
-		 * ignored.
+		 * handler. Android only support maps/key String value pairs. Other JSON objects
+		 * will be rejected.
 		 */
 		public Builder data(String data) {
 			return new Builder(this.alert, this.badge, this.sound, this.priority, data, this.sendAsSms, this.apns,
 					this.gcm, this.wns, this.baidu, this.w3c, this.custom);
 		}
 
+		/**
+		 * The data is a key/value map. The object is not shown in the notification, but
+		 * can be used by the mobile app notification handler. Android only support
+		 * maps/key String value pairs. Other JSON objects will be rejected.
+		 * 
+		 * @throws JsonProcessingException
+		 */
+		public Builder data(Map<String, Object> data) throws JsonProcessingException {
```

Comment:
You could actually add another convenience method to accept any java object to be serialized into json (then supports jackson annotations like `@JsonIgnore`).

Reply:
Good point, but using Object instead of Map works fine too ;-)